src/Microsoft.AspNet.OData.Shared/Builder/ODataModelBuilder.cs

```diff
@@ -44,7 +45,23 @@ public ODataModelBuilder()
         /// <summary>
         /// Gets or sets the namespace that will be used for the resulting model
         /// </summary>
-        public string Namespace { get; set; }
+        public string Namespace
+        {
+            get
+            {
+                return _namespace;
+            }
+            set
+            {
+                if (String.IsNullOrEmpty(value))
+                {
+                    throw Error.PropertyNull();
+                }
```

Comment:
I generally dislike having an initial state that can't be returned to.

Instead of throwing, perhaps we should set the namespace back to "Default" (and set HasAssignedNamespace back to false)?

Reply:
Default namespace is being set to "Default". This check is just to ensure that you can't set the value to null.

Reply:
I understand that the user can set the namespace to "Default", but, from the code perspective, this is (subtly) different (though currently functionality the same) than never having set it in the first place. I was wondering if setting the namespace to null shouldn't reset to the default state (set HasAssignedNamespace to false and namespace to "Default", rather than throw an error.

This is more of a philosophical thing; we have an initial state (HasAssignedNamespace) that can't be reset.  You can mimic the behavior, by knowing that the default namespace is "Default", and explicitly setting that, but there's no way to fully "undo" the fact that you touched this property.  So, for example, if we added other behavior based on whether the namespace was explicitly set, the developer would have no way to get back to that state.

---
In reply to: [234746322](https://github.com/OData/WebApi/pull/1592#discussion_r234746322) [](ancestors = 234746322)

Reply:
Sure makes sense. Removing the error.

Reply:
Done.